src/main/java/org/group23/ControllerStructure.java

```diff
@@ -1,55 +1,108 @@
+    public String addQuestion(
+            @PathVariable Long surveyId,
+            @ModelAttribute("survey") Survey survey,
+            @RequestParam String questionText,
+            Model model
+    ) {
+        // Get the survey by ID to ensure we have the latest data
+        Survey updatedSurvey = surveyRepository.findById(surveyId).orElse(null);
+        if (updatedSurvey != null) {
+            // Create a text question and add it to the survey
+            TextQuestion textQuestion = new TextQuestion(questionText);
+            updatedSurvey.addQuestion(textQuestion);
+            surveyRepository.save(updatedSurvey);
 
-    @PostMapping("/createSurvey")
-    public String createSurvey(Model model){
-        Survey survey = new Survey("Survey 1");
-        ArrayList<Question> questions = new ArrayList<>();
-        TextQuestion question = new TextQuestion("What's your name?");
-        question.addAnswer("Jeff");
-        question.addAnswer("Jimmy");
-        questions.add(question);
-        questions.add(new TextQuestion("What's your favorite color?"));
-        survey.setQuestions(questions);
-        surveyRepository.save(survey);
-        model.addAttribute("message", "Hello, World!");
-        return "helloPage";
+            model.addAttribute("survey", updatedSurvey);
+
+            return "redirect:/addRemoveQuestions/" + surveyId;
+        } else {
+            // Handle survey not found
+            return "error";
+        }
+    }
+
+    @GetMapping("/deleteQuestion/{surveyId}/{questionId}")
+    public String deleteQuestion(
+            @PathVariable Long surveyId,
+            @PathVariable Long questionId
+    ) {
+        // Get the survey by ID
+        Survey survey = surveyRepository.findById(surveyId).orElse(null);
+        if (survey != null) {
+            // Find and remove the question from the survey
+            Question questionToRemove = survey.getQuestions().stream()
+                    .filter(question -> question.getId().equals(questionId))
+                    .findFirst()
+                    .orElse(null);
+
+            if (questionToRemove != null) {
+                survey.removeQuestion(questionToRemove);
+                surveyRepository.save(survey);
+            }
+            return "redirect:/addRemoveQuestions/" + survey.getId();
+        } else {
+            // Handle survey not found
+            return "error";
+        }
     }
 
-    @GetMapping("/getSurvey")
-    public String getSurvey(Model model){
-        Survey survey = surveyRepository.findByName("Survey 1");
-        //System.out.println(survey);
-        model.addAttribute("message", "Hello, World!");
-        return "helloPage";
+    @GetMapping("/saveSurvey")
+    public String saveSurvey() {
+        return "redirect:/surveyCreated";
+    }
+
+    @GetMapping("/surveyCreated")
+    public String surveyCreated(Model model) {
+        Survey survey = surveyRepository.findById(1L).orElse(null);
```

Comment:
This should try to get whichever survey was created, not the one with ID 1.

Reply:
done